pwiz_tools/Skyline/TestFunctional/EADZIonsTest.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System.Globalization;
+using System.Linq;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using pwiz.Common.SystemUtil;
+using pwiz.Skyline.Controls.SeqNode;
+using pwiz.Skyline.Model;
+using pwiz.Skyline.Properties;
+using pwiz.Skyline.SettingsUI;
+using pwiz.Skyline.Util;
+using pwiz.SkylineTestUtil;
+
+namespace pwiz.SkylineTestFunctional
+{
+    /// <summary>
+    /// Testing z+1 and z+2 ions feature.
+    /// </summary>
+    [TestClass]
+    public class EADZIonsTest : AbstractFunctionalTestEx
+    {
+        [TestMethod]
+        public void TestEADIons()
+        {
+            Run(@"TestFunctional\EADZIonsTest.zip");
+        }
+
+        protected override void DoTest()
+        {
+            //Making sure the input aliases are unique across all ion types to avoid input ambiguity.
+            Assert.IsTrue(Enumerable.Range(0, (int) IonType.zhh + 1).SelectMany(i => ((IonType) i).GetInputAliases(), (i, s) => s)
+                .GroupBy(s => s, (s, enumerable) => enumerable.Count()).All(i => i == 1));
+
+            OpenDocument("EADZIonsTest.sky");
+            var transitionSettingsUI = ShowDialog<TransitionSettingsUI>(SkylineWindow.ShowTransitionSettingsUI);
+            RunUI(() =>
+            {
+                transitionSettingsUI.FragmentTypes = "c, z., z'";
+                transitionSettingsUI.IonCount = 5;
+            });
+            OkDialog(transitionSettingsUI, transitionSettingsUI.OkDialog);
+            WaitForGraphs();
+            var libMatch = SkylineWindow.GraphSpectrum.DisplayedSpectrum;
+            ImportResults("FilteredScans\\LITV56_EAD" + ExtensionTestContext.ExtMzml);
+            WaitForGraphs();
+            FindNode((505.5810).ToString("F4", LocalizationHelper.CurrentCulture) + "+++");
+            Settings.Default.ShowZHIons = true;
+            Settings.Default.ShowZHHIons = true;
+
+            var testIons = new[]{
+                new {type = IonType.zh, offset = 8},
+                new {type = IonType.zhh, offset = 5},
+                new {type = IonType.zhh, offset = 5}

```

Comment:
Is there a reason that "type = IonType.zhh, offset = 5" appears in this list twice?

Reply:
Yes, it was supposed to be `type = IonType.zh, offset = 5`